File: source/frontend/src/app/entities/vacature/adresHouding.ts
import {AdresBuitenland} from './adresBuitenland';
import {AdresNederland} from './adresNederland';
import {ModelDate} from '../shared/modelDate';


export interface AdresHouding {
  codeFunctieAdres?: string;
  datumAanvangAdres?: ModelDate;
  datumEindeAdres?: ModelDate;
  adres?: AdresNederland | AdresBuitenland;
}

export function isAdresNederland(object: any): object is AdresNederland {
  return 'adresNederland' in object;
}

File: source/backend-vacatures-bemiddelaar/src/main/java/com/cap/pocvng/entity/AdresHouding.java
package com.cap.pocvng.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.Valid;
import javax.validation.constraints.Size;
import java.time.LocalDate;

/**
 * AdresHouding
 */


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AdresHouding {

    @Size(max = 1)
    private String codeFunctieAdres;

    private LocalDate datumAanvangAdres;

    private LocalDate datumEindeAdres;

    @Valid
    private Adres adres;

}
